app/models/secret_storage.rb

```diff
@@ -50,6 +50,51 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    VAULT_SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(vault_path(key))
+      return false if result.nil?
+      result = result.dup.to_h
+      # making sure that the delete won't blow up
+      result[:value] = result[:data][:vault]
+      result
+    end
+
+    def self.write(key, data)
+      Vault.logical.write(vault_path(key), vault: data[:value])
+    end
+
+    def self.delete(key)
+      Vault.logical.delete(vault_path(key))
+    end
+
+    def self.keys()
+      Vault.logical.list(VAULT_SECRET_BACKEND).map! { |key| encode_path(key, :decode) }
+    end
+
+    private
+
+    def self.vault_path(key)
+      VAULT_SECRET_BACKEND + encode_path(key)
+    end
+
+    def self.encode_path(string, direction = :encode)
```

Comment:
call this covert_path and make the direction mandatory ? `covert_path key, :encode` + `covert_path key, :decode`